File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> map;
        int lead = 0;
        for(auto i : nums) map[i]++;
        for(auto i : map){
            if(i.second > nums.size()/2) lead = i.first;
        }
        return lead;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int candidate = 0, lead = 0;
        for(int i=0; i<nums.length; i++)
        {
            if(lead == 0)
                candidate = nums[i];
            if(candidate == nums[i])
                lead++;
            else
                lead--;
        }
        return candidate;
    }
}